src/NetMQ/Core/Patterns/XPub.cs

```diff
@@ -176,15 +184,29 @@ protected override void XReadActivated(Pipe pipe)
                         // If the subscription is not a duplicate, store it so that it can be
                         // passed to used on next recv call.
                         if (m_options.SocketType == ZmqSocketType.Xpub && (unique || m_verbose))
-                            m_pending.Enqueue(sub.CloneData());
+                        {
+                            m_pending.Enqueue(sub);
+                        }
+                        else
+                        {
+                            sub.Close();
+                        }
+
                     }
                 }
+                else if (m_broadcastEnabled && size > 0 && sub.HasMore && sub[0] == 2)
+                {
+                    m_lastPipe = pipe;
+                    m_lastPipeIsBroadcast = true;
+                    sub.Offset = sub.Offset + 1;
+                    sub.Size = sub.Size - 1;
+                    m_pending.Enqueue(sub);
```

Comment:
This is a good point! But the we need to change the match logic in XSend so that it ignores the first byte in the first frame. I will remove the check for .HasMore to support a single frame case. And will change the match logic.